### PR TITLE
Handle middleware passed after the service to app.use

### LINKED ISSUE #176: Feature request: (req, res, next) handler for *after* the service
Looks like this works:
```js
  app.use(endpoint, function (req, res, next) {
    console.log(`${endpoint} before 1`);
    next();
  }, function (req, res, next) {
    console.log(`${endpoint} before 2`);
    next();
  }, Stream);
```

But this doesn't: (`TypeError: Router.use() requires middleware function but got a Object`)

```js
  app.use(endpoint, function (req, res, next) {
    console.log(`${endpoint} before 1`);
    next();
  }, function (req, res, next) {
    console.log(`${endpoint} before 2`);
    next();
  }, Stream, function (req, res, next) {
    console.log(`${endpoint} after`);
    next();
  });
```

I also tried this, but the `after` handler simply gets ignored:
```js
  app.use(endpoint, function (req, res, next) {
    console.log(`${endpoint} before 1`);
    next();
  }, function (req, res, next) {
    console.log(`${endpoint} before 2`);
    next();
  }, Stream);
  app.use(endpoint, function (req, res, next) {
    console.log(`${endpoint} after`);
    next();
  });
```

### PR DESCRIPTION
**Pull request by dbkaplun**
Closes #176

### CODE REVIEW

File: lib/application.js [hunk 1]
```diff
@@ -59,7 +59,9 @@ module.exports = {
   use: function () {
     var args = _.toArray(arguments);
     var location = args.shift();
-    var service = args.pop();
+    var parts = _.partition(args, _.isFunction);
+    if (parts[1].length > 1) throw new Error('more than one service passed to app.use');
```
Comment (daffl) [inline lib/application.js]: Small style nit, I'd make it

```js
if (parts[1].length > 1) {
  throw new Error('more than one service passed to app.use');
}
```

File: lib/application.js [hunk 2]
```diff
@@ -73,7 +75,7 @@ module.exports = {
 
     this.service(location, service, {
       // Any arguments left over are other middleware that we want to pass to the providers
-      middleware: args
+      middleware: parts[0]
```
Comment (daffl) [inline lib/application.js]: I think this would just add everything before the service happens right? Maybe we need a `middleware: { before: [], after: [] }`?

**Comment by daffl** [comment]
Awesome, thank you! I made some comments and it would be nice to have some tests for this functionality (probably similar to https://github.com/feathersjs/feathers/blob/master/test/providers/rest.test.js#L337).
After that we can definitely get it into the new release!

**Comment by dbkaplun** [comment]
@daffl fixed and added tests

File: test/application.test.js [hunk 3]
```diff
@@ -128,21 +128,25 @@ describe('Feathers application', function () {
     var app = feathers()
       .configure(feathers.rest())
       .use('/todo', function (req, res, next) {
-        req.feathers.stuff = 'custom middleware';
+        req.feathers.preService = 'pre-service middleware';
         next();
-      }, todoService)
+      }, todoService, function (req, res, next) {
+        res.set('post-service', res.data.id);
```
Comment (daffl) [inline test/application.test.js]: I think we could set and then check for `res.data.postService` instead since the handler should run last.

**Comment by daffl** [comment]
This is great! Thank you! One more small comment for the test. Nothing major but then we can get it in and ready for a version 1.3.

File: test/application.test.js (hunk 3 above)
Comment (dbkaplun) [inline test/application.test.js]: Sorry I don't understand, can you rephrase please?

Comment (daffl) [inline test/application.test.js]: What I mean is to instead set something like:

```js
res.data.postService = true;
```

and then in the test check for

```js
assert.ok(data.postService, 'Post-service middleware updated response');
```

instead of `assert.equal(response.headers['post-service'], 'dishes', 'Post-service middleware updated response');`.

Comment (dbkaplun) [inline test/application.test.js]: I see! I did it this way because I wanted to actually ensure that the post-service handler actually runs after the service, but I can change it if you like.

Comment (daffl) [inline test/application.test.js]: Good point. Lets leave it then. I will merge shortly. Thanks again!

**Comment by ekryski** [comment]
Oh nice! This PR snuck by me but I've run into this before. Thanks @dbkaplun!
